### ISSUE TITLE
Reporting of Forth Characteristics

### ISSUE THREAD

**Issue by steverpalmer** [gerryjackson/forth2012-test-suite#2]
While running the tests, certain Implementation-defined characteristics of the Forth system may be derived, such as the number of bits in a cell.  It would be nice if these derived characteristics of the Forth system could be reported along with the test results.  (This is a similar to the behaviour of the floating point Paranoia - http://www.netlib.org/paranoia/.)

**Comment by steverpalmer** [comment]
For example, I commented out several test suites in the runtests.fth file, and then ran the tests and got the following output...

    ---------------------------
            Error Report
    Word Set             Errors
    ---------------------------
    Core                    -
    Core extension          -
    Double number           -
    Exception               -
    Facility                -
    File-access             -
    Locals                  -
    Memory-allocation       1
    Programming-tools       -
    Search-order            -
    String                  -
    ---------------------------
    Total                   1
    ---------------------------

There is no indication that the Double number, Locals, Programming-tools and String tests where not run.

**Comment by steverpalmer** [comment]
For example, when running the tests, included in the details are the lines:

    YOU SHOULD SEE THE NUMBER RANGES OF SIGNED AND UNSIGNED NUMBERS:
      SIGNED: -8000000000000000 7FFFFFFFFFFFFFFF 
    UNSIGNED: 0 FFFFFFFFFFFFFFFF 

This is potentially valuable information, but could be easily missed in amongst the other output.  It would be nicer if there was a separate section in the output, perhaps near the Error Report at the end where this information could be collected and displayed.

**Comment by gerryjackson** [comment]
Good idea. We need to define what characteristics could be reported. Number ranges are obvious as are bits/cell and au's/char. We could report results from environment queries which are otherwise untested. Let's get the next release of the test suite sorted out first.

**Comment by steverpalmer** [comment]
I agree that this is not an urgent issue, but perhaps this "issue" could be used to catch any ideas we have about characteristic reporting.

One other idea that I had was that the test suite could also list words that are supplied but are considered obsolete, such as #TIB, CONVERT, EXPECT, QUERY, SPAN, TIB, etc.  This may be particularly relevant if these words are not tested.

**Comment by gerryjackson** [comment]
I deliberately didn't include words marked obsolescent in the original test programs as I hadn't included them in my system. Nobody complained. In Forth 2012 [COMPILE] was designated obsolescent so I removed any tests on that - Stephen Pelc of MPE had complained when his system failed and suggested using POSTPONE instead which would have been a curious way of testing [COMPILE] :) I'm open to having tests on obsolete words but would suggest they went in a separate test program.

**Comment by gerryjackson** [comment]
Issue #18 suggests another item that could be reported, the behaviour of ALIGN and ALIGNED.
